python/core/qgsstatisticalsummary.sip

```diff
@@ -1,214 +1,299 @@
+%Docstring
+ Calculator for summary statistics for a list of doubles.
+
+ Statistics are calculated by calling calculate() and passing a list of doubles. The
+ individual statistics can then be retrieved using the associated methods. Note that not all statistics
+ are calculated by default. Statistics which require slower computations are only calculated by
+ specifying the statistic in the constructor or via setStatistics().
+
+.. versionadded:: 2.9
 %End
 
+%TypeHeaderCode
+#include "qgsstatisticalsummary.h"
+%End
   public:
 
-    //! Enumeration of flags that specify statistics to be calculated
     enum Statistic
     {
-      Count,  //!< Count
-      CountMissing, //!< Number of missing (null) values
-      Sum,  //!< Sum of values
-      Mean,  //!< Mean of values
-      Median, //!< Median of values
-      StDev, //!< Standard deviation of values
-      StDevSample, //!< Sample standard deviation of values
-      Min,  //!< Min of values
-      Max,  //!< Max of values
-      Range, //!< Range of values (max - min)
-      Minority, //!< Minority of values
-      Majority, //!< Majority of values
-      Variety, //!< Variety (count of distinct) values
-      FirstQuartile, //!< First quartile
-      ThirdQuartile, //!< Third quartile
-      InterQuartileRange, //!< Inter quartile range (IQR)
+      Count,
+      CountMissing,
+      Sum,
+      Mean,
+      Median,
+      StDev,
+      StDevSample,
+      Min,
+      Max,
+      Range,
+      Minority,
+      Majority,
+      Variety,
+      FirstQuartile,
+      ThirdQuartile,
+      InterQuartileRange,
       All
     };
     typedef QFlags<QgsStatisticalSummary::Statistic> Statistics;
 
-    /** Constructor for QgsStatisticalSummary
-     * @param stats flags for statistics to calculate
-     */
-    QgsStatisticalSummary( QgsStatisticalSummary::Statistics stats = QgsStatisticalSummary::All );
+
+    QgsStatisticalSummary( QgsStatisticalSummary::Statistics stats = All );
+%Docstring
+ Constructor for QgsStatisticalSummary
+ \param stats flags for statistics to calculate
+%End
 
     virtual ~QgsStatisticalSummary();
 
-    /** Returns flags which specify which statistics will be calculated. Some statistics
-     * are always calculated (e.g., sum, min and max).
-     * @see setStatistics
-     */
-    QgsStatisticalSummary::Statistics statistics() const;
-
-    /** Sets flags which specify which statistics will be calculated. Some statistics
-     * are always calculated (e.g., sum, min and max).
-     * @param stats flags for statistics to calculate
-     * @see statistics
-     */
-    void setStatistics( QgsStatisticalSummary::Statistics stats );
-
-    /** Resets the calculated values
-     */
+    Statistics statistics() const;
+%Docstring
+ Returns flags which specify which statistics will be calculated. Some statistics
+ are always calculated (e.g., sum, min and max).
+.. seealso:: setStatistics
+ :rtype: Statistics
+%End
+
+    void setStatistics( Statistics stats );
```

Comment:
you might need to add `QgsStatisticalSummary`in front of `Statistics` (in the header)